config file changes for flow dump tests

anupama-dasari opened the pull request:
Signed-off-by: Anupama Dasari anupama.dasari@intel.com

hirentha commented:
@anupama-dasari  Please attach sucessful ran logs with this PR. And please explain what changes you did in test_to_run.txt, I didn't understnd that changes. Thanks!!

anupama-dasari commented:
> @anupama-dasari Please attach sucessful ran logs with this PR. And please explain what changes you did in test_to_run.txt, I didn't understnd that changes. Thanks!!

Modified config files DPDK_Flow_Dump_with_LNT.json and DPDK_Flow_Dump_with_PNA_indirect_counter.json as the flow dump table entries verification was failing. 

Also re-sequenced the tests in tests_to_run.txt as LNT_ECMP_2HotPlugVM_2Host_netperf and LNT_FRR_with_ECMP_netperf_LinkFlap were always failing in the full run and passing with re-run. So moved these 2 tests to the end of the test suite. 

Pass Logs:

[root@ipdk-bdc01 p4-ptf]# bash -c 'python ipdk_test_runner.py -f tests_to_run_anupama.txt -s /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install -o /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe -d /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps -vm /home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_1.img,/home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_2.img,/home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_3.img,/home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_4.img -lnt_bdf 0000:c1:00.0,0000:c1:00.1 -bdf 0000:81:00.0 -client '\''10.138.167.142,root,root@123'\'' -port ens4f0,ens4f1 --log_file ptf_tests.log --verbose'
replacing VM1 with /home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_1.img
replacing VM2 with /home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_2.img
replacing VM3 with /home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_3.img
replacing VM4 with /home/admin/VM_Images/ubuntu-20.04-server-cloudimg-amd64_4.img
replacing BDF1 with 0000:81:00.0
replacing LNT_BDF1 with 0000:c1:00.0
replacing LNT_BDF2 with 0000:c1:00.1
replacing PORT1 with ens4f0
replacing PORT2 with ens4f1

Running DPDK_Flow_Dump_with_LNT

2023-07-13T04:14:16Z|00001|vlog|INFO|opened log file /tmp/ovs-vswitchd.log
2023-07-13T04:14:16Z|00002|ovs_numa|INFO|Discovered 16 CPU cores on NUMA node 0
2023-07-13T04:14:16Z|00003|ovs_numa|INFO|Discovered 1 NUMA nodes and 16 CPU cores
2023-07-13T04:14:16Z|00004|reconnect|INFO|unix:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/var/run/openvswitch/db.sock: connecting...
2023-07-13T04:14:16Z|00005|reconnect|INFO|unix:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/var/run/openvswitch/db.sock: connected
/usr/local/bin/ptf:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/usr/local/lib/python3.9/site-packages/scapy/layers/ipsec.py:471: CryptographyDeprecationWarning: Blowfish has been deprecated
  cipher=algorithms.Blowfish,
/usr/local/lib/python3.9/site-packages/scapy/layers/ipsec.py:485: CryptographyDeprecationWarning: CAST5 has been deprecated
  cipher=algorithms.CAST5,
DPDK_Flow_Dump_with_LNT.DPDK_Flow_Dump_with_LNT ... I20230713 09:44:20.222393 1426866 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:20.348327 1426876 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:20.531782 1426884 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:20.749449 1426895 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:20.879604 1426905 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.006323 1426914 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.134445 1426925 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.264779 1426935 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.390010 1426945 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.517062 1426955 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.641288 1426965 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.764871 1426975 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:21.891500 1426985 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.017011 1426995 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.140398 1427005 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.267484 1427015 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.394177 1427025 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.519022 1427035 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.642990 1427045 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.768883 1427055 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:22.896724 1427065 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.021883 1427075 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.146459 1427085 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.276515 1427095 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.401583 1427105 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.529599 1427115 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.653563 1427125 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.782564 1427135 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:23.910523 1427145 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.034502 1427155 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.163539 1427165 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.290904 1427175 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.421417 1427185 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.548375 1427195 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.675997 1427205 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.801522 1427215 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:24.929744 1427225 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:25.053647 1427235 gnmi_ctl.cc:103] Client context cancelled.
common/p4c_artifacts/linux_networking/linux_networking.p4(100): [--Wwarn=unused] warning: Table always_recirculate_table is not used; removing
    table always_recirculate_table {
          ^^^^^^^^^^^^^^^^^^^^^^^^
common/p4c_artifacts/linux_networking/linux_networking.p4(493): [--Wwarn=unused] warning: Table handle_rx_exception_pkts is not used; removing
    table handle_rx_exception_pkts {
          ^^^^^^^^^^^^^^^^^^^^^^^^
common/p4c_artifacts/linux_networking/linux_networking.p4(533): [--Wwarn=unused] warning: Table handle_tx_exception_pkts is not used; removing
    table handle_tx_exception_pkts {
          ^^^^^^^^^^^^^^^^^^^^^^^^
common/p4c_artifacts/linux_networking/linux_networking.p4(313): [--Wwarn=invalid_header] warning: accessing a field of a potentially invalid header hdr.outer_ethernet
        hdr.outer_ethernet.dst_addr = dst_mac_addr;
        ^^^^^^^^^^^^^^^^^^
common/p4c_artifacts/linux_networking/linux_networking.p4(295): [--Wwarn=invalid_header] warning: accessing a field of a potentially invalid header hdr.outer_ethernet
        hdr.outer_ethernet.src_addr = src_mac_addr;
        ^^^^^^^^^^^^^^^^^^
[--Wwarn=mismatch] warning: Mismatched header/metadata struct for key elements in table ipv4_tunnel_term_table. Copying all match fields to metadata
[--Wwarn=mismatch] warning: Mismatched header/metadata struct for key elements in table as_ecmp_sel. Copying all match fields to metadata
[--Wwarn=mismatch] warning: Mismatched header/metadata struct for key elements in table handle_tx_control_vlan_pkts_table. Copying all match fields to metadata
I20230713 09:44:31.329943 1427270 tdi_pipeline_builder.cc:116] Found P4 program: linux_networking
I20230713 09:44:31.330284 1427270 tdi_pipeline_builder.cc:123]  Found pipeline: pipe
ok

----------------------------------------------------------------------
Ran 1 test in 23.532s

OK
Killing qemu
killing infrap4d
killing 1421975
Sleeping for 2 seconds
Killing ovs
sleeping for 2 seconds
Removing any vhost users from /tmp
Setting PATH
OS and Version details...
Fedora : 33


Updated Environment Variables ...
SDE_INSTALL: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install
LIBRARY_PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:
LD_LIBRARY_PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib/:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/usr/local/lib:/usr/local/lib64
PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/sbin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/sbin:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/sbin

starting ovs
set hugepages
2048
2048
run infrap4d
check if infrap4d is running
root 1426823 1 0 09:44 ? 00:00:00 /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/sbin/infrap4d
infrap4d Running
Using packet manipulation module: ptf.packet_scapy
7/13/2023,9:44:20  INFO: Begin to configure OVVS and VM on local host
7/13/2023,9:44:20  PASSED: gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host0,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-0,packet-dir:host,port-type:LINK"
7/13/2023,9:44:20  PASSED: gnmi-ctl set "device:virtual-device,name:net_vhost1,host-name:host1,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-1,packet-dir:host,port-type:LINK"
7/13/2023,9:44:20  PASSED: gnmi-ctl set "device:physical-device,name:PORT0,pci-bdf:0000:c1:00.0,control-port:TAP1,mtu:1500,packet-dir:network,port-type:LINK"
7/13/2023,9:44:20  PASSED: gnmi-ctl set "device:physical-device,name:PORT1,pci-bdf:0000:c1:00.1,control-port:TAP2,mtu:1500,packet-dir:network,port-type:LINK"
7/13/2023,9:44:20  PASSED: gnmi-ctl set "device:virtual-device,name:TAP0,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,packet-dir:host,port-type:TAP"
7/13/2023,9:44:21  PASSED: gnmi-ctl set "device:virtual-device,name:TAP3,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,packet-dir:host,port-type:TAP"
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:virtual-device,name:net_vhost0,host-name:host0,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-0,packet-dir:host,port-type:LINK
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:virtual-device,name:net_vhost1,host-name:host1,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-1,packet-dir:host,port-type:LINK
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:physical-device,name:PORT0,pci-bdf:0000:c1:00.0,control-port:TAP1,mtu:1500,packet-dir:network,port-type:LINK
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:physical-device,name:PORT1,pci-bdf:0000:c1:00.1,control-port:TAP2,mtu:1500,packet-dir:network,port-type:LINK
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:virtual-device,name:TAP0,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,packet-dir:host,port-type:TAP
7/13/2023,9:44:25  PASSED: gnmi-ctl get verified for device:virtual-device,name:TAP3,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,packet-dir:host,port-type:TAP
7/13/2023,9:44:25  PASSED: ip  link set TAP0 up
7/13/2023,9:44:25  PASSED: succeed to enable TAP0 up
7/13/2023,9:44:25  PASSED: ip  link set TAP1 up
7/13/2023,9:44:25  PASSED: succeed to enable TAP1 up
7/13/2023,9:44:25  PASSED: ip  addr add 40.1.1.1/24 dev TAP1
7/13/2023,9:44:25  INFO: p4c-dpdk -I p4include -I p4include/dpdk --p4v=16 --p4runtime-files             common/p4c_artifacts/linux_networking/p4Info.txt -o common/p4c_artifacts/linux_networking/pipe/linux_networking.spec --arch pna --bf-rt-schema common/p4c_artifacts/linux_networking/bf-rt.json --context             common/p4c_artifacts/linux_networking/pipe/context.json common/p4c_artifacts/linux_networking/linux_networking.p4
7/13/2023,9:44:31  PASSED: p4c-dpdk -I p4include -I p4include/dpdk --p4v=16 --p4runtime-files             common/p4c_artifacts/linux_networking/p4Info.txt -o common/p4c_artifacts/linux_networking/pipe/linux_networking.spec --arch pna --bf-rt-schema common/p4c_artifacts/linux_networking/bf-rt.json --context             common/p4c_artifacts/linux_networking/pipe/context.json common/p4c_artifacts/linux_networking/linux_networking.p4
7/13/2023,9:44:31  PASSED: ovs_pipeline_builder --p4c_conf_file=linux_networking.conf             --bf_pipeline_config_binary_file=linux_networking.pb.bin
7/13/2023,9:44:32  PASSED: p4rt-ctl set pipe: common/p4c_artifacts/linux_networking/linux_networking.pb.bin
7/13/2023,9:44:32  PASSED: Successfully added bridge to ovs
7/13/2023,9:44:32  PASSED: ip  link set br-int up
7/13/2023,9:44:32  PASSED: succeed to enable br-int up
7/13/2023,9:44:32  INFO: Config VXLAN Port on local host With Remote Tunnel IP
7/13/2023,9:44:33  PASSED: Successfully added vxlan port to ovs
7/13/2023,9:44:33  PASSED: ip  link add link TAP0 name vlan1 type vlan id 1
7/13/2023,9:44:33  PASSED: succeed to add vlan1 to port TAP0
7/13/2023,9:44:33  PASSED: Successfully added vlan port to ovs
7/13/2023,9:44:33  PASSED: ip  link set vlan1 up
7/13/2023,9:44:33  PASSED: succeed to enable vlan1 up
7/13/2023,9:44:33  PASSED: ip  link add link TAP0 name vlan2 type vlan id 2
7/13/2023,9:44:33  PASSED: succeed to add vlan2 to port TAP0
7/13/2023,9:44:33  PASSED: Successfully added vlan port to ovs
7/13/2023,9:44:33  PASSED: ip  link set vlan2 up
7/13/2023,9:44:33  PASSED: succeed to enable vlan2 up
7/13/2023,9:44:33  INFO: Scenario : lnt_tx_control_pkts_table
7/13/2023,9:44:33  INFO: Adding lnt_tx_control_pkts_table rules
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=0,action=linux_networking_control.push_vlan_fwd(6,1)
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=1,action=linux_networking_control.push_vlan_fwd(6,2)
7/13/2023,9:44:33  INFO: Scenario : lnt_tx_control_vlan_pkts_table
7/13/2023,9:44:33  INFO: Adding lnt_tx_control_vlan_pkts_table rules
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=6,local_metadata.vlan_id=1,action=linux_networking_control.pop_vlan_fwd(0)
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=6,local_metadata.vlan_id=2,action=linux_networking_control.pop_vlan_fwd(1)
7/13/2023,9:44:33  INFO: Scenario : lnt_control_physical_port_rx_table
7/13/2023,9:44:33  INFO: Adding lnt_control_physical_port_rx_table rules
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=2,action=linux_networking_control.set_control_dest(3)
7/13/2023,9:44:33  INFO: Scenario : lnt_control_physical_tx_table
7/13/2023,9:44:33  INFO: Adding lnt_control_physical_tx_table rules
7/13/2023,9:44:33  PASSED: p4rt-ctl add entry: istd.input_port=3,action=linux_networking_control.set_control_dest(2)
7/13/2023,9:44:43  INFO: Get flow dump table entry
['Table entries for bridge br0:', '  table=linux_networking_control.vxlan_encap_mod_table vendormeta_mod_data_ptr=0x000000 actions=linux_networking_control.vxlan_encap(src_addr=0x28010101dst_addr=0x28010102dst_port=0x12b5vni=0x000000)', '  table=linux_networking_control.ipv4_tunnel_term_table tunnel_type=0x02,ipv4_src=0x28010102,ipv4_dst=0x28010101 actions=linux_networking_control.decap_outer_ipv4(tunnel_id=0x000000)', '  table=linux_networking_control.handle_rx_control_pkts_table istd.input_port=0x00000002 actions=linux_networking_control.set_control_dest(port_id=0x00000003)', '  table=linux_networking_control.handle_tx_control_vlan_pkts_table istd.input_port=0x00000006,local_metadata.vlan_id=0x0001 actions=linux_networking_control.pop_vlan_fwd(port=0x00000000)', '  table=linux_networking_control.handle_tx_control_vlan_pkts_table istd.input_port=0x00000006,local_metadata.vlan_id=0x0002 actions=linux_networking_control.pop_vlan_fwd(port=0x00000001)', '  table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000000 actions=linux_networking_control.push_vlan_fwd(port=0x00000006vlan_tag=0x0001)', '  table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000001 actions=linux_networking_control.push_vlan_fwd(port=0x00000006vlan_tag=0x0002)', '  table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000003 actions=linux_networking_control.set_control_dest(port_id=0x00000002)', '']
7/13/2023,9:44:43  INFO: Verify each entry
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.vxlan_encap_mod_table vendormeta_mod_data_ptr=0x000000 actions=linux_networking_control.vxlan_encap(src_addr=0x28010101dst_addr=0x28010102dst_port=0x12b5vni=0x000000)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.ipv4_tunnel_term_table tunnel_type=0x02,ipv4_src=0x28010102,ipv4_dst=0x28010101 actions=linux_networking_control.decap_outer_ipv4(tunnel_id=0x000000)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_rx_control_pkts_table istd.input_port=0x00000002 actions=linux_networking_control.set_control_dest(port_id=0x00000003)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_tx_control_vlan_pkts_table istd.input_port=0x00000006,local_metadata.vlan_id=0x0001 actions=linux_networking_control.pop_vlan_fwd(port=0x00000000)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_tx_control_vlan_pkts_table istd.input_port=0x00000006,local_metadata.vlan_id=0x0002 actions=linux_networking_control.pop_vlan_fwd(port=0x00000001)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000000 actions=linux_networking_control.push_vlan_fwd(port=0x00000006vlan_tag=0x0001)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000001 actions=linux_networking_control.push_vlan_fwd(port=0x00000006vlan_tag=0x0002)" in flow dump is verified
7/13/2023,9:44:43  PASSED: The entry "table=linux_networking_control.handle_tx_control_pkts_table istd.input_port=0x00000003 actions=linux_networking_control.set_control_dest(port_id=0x00000002)" in flow dump is verified
7/13/2023,9:44:43  INFO: Begin to teardown ...
7/13/2023,9:44:43  INFO: Delete match action rules on local host
7/13/2023,9:44:43  INFO: Deleting lnt_tx_control_pkts_table rules
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=0
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=1
7/13/2023,9:44:43  INFO: Deleting lnt_tx_control_vlan_pkts_table rules
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=6,local_metadata.vlan_id=1
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=6,local_metadata.vlan_id=2
7/13/2023,9:44:43  INFO: Deleting lnt_control_physical_port_rx_table rules
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=2
7/13/2023,9:44:43  INFO: Deleting lnt_control_physical_tx_table rules
7/13/2023,9:44:43  PASSED: p4rt-ctl del entry: istd.input_port=3
7/13/2023,9:44:43  INFO: Delete vlan on local host
7/13/2023,9:44:43  PASSED: ip  link del vlan1
7/13/2023,9:44:43  PASSED: succeed to delee vlan1
7/13/2023,9:44:43  PASSED: ip  link del vlan2
7/13/2023,9:44:43  PASSED: succeed to delee vlan2
7/13/2023,9:44:43  INFO: Test has PASSED


Running DPDK_Flow_Dump_with_PNA_indirect_counter

2023-07-13T04:14:50Z|00001|vlog|INFO|opened log file /tmp/ovs-vswitchd.log
2023-07-13T04:14:50Z|00002|ovs_numa|INFO|Discovered 16 CPU cores on NUMA node 0
2023-07-13T04:14:50Z|00003|ovs_numa|INFO|Discovered 1 NUMA nodes and 16 CPU cores
2023-07-13T04:14:50Z|00004|reconnect|INFO|unix:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/var/run/openvswitch/db.sock: connecting...
2023-07-13T04:14:50Z|00005|reconnect|INFO|unix:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/var/run/openvswitch/db.sock: connected
/usr/local/bin/ptf:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/usr/local/lib/python3.9/site-packages/scapy/layers/ipsec.py:471: CryptographyDeprecationWarning: Blowfish has been deprecated
  cipher=algorithms.Blowfish,
/usr/local/lib/python3.9/site-packages/scapy/layers/ipsec.py:485: CryptographyDeprecationWarning: CAST5 has been deprecated
  cipher=algorithms.CAST5,
DPDK_Flow_Dump_with_PNA_indirect_counter.DPDK_Flow_Dump_with_indirect_counter ... I20230713 09:44:53.798530 1427801 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:53.924054 1427811 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.052973 1427821 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.179756 1427831 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.303843 1427841 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.430440 1427851 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.556399 1427861 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.683564 1427871 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.807653 1427881 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:54.931452 1427891 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:55.058763 1427901 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:55.184474 1427911 gnmi_ctl.cc:103] Client context cancelled.
I20230713 09:44:56.805366 1427937 tdi_pipeline_builder.cc:116] Found P4 program: l3em_indirect_counter_pna
I20230713 09:44:56.805555 1427937 tdi_pipeline_builder.cc:123]  Found pipeline: pipe
ok

----------------------------------------------------------------------
Ran 1 test in 3.721s

OK
Killing qemu
killing infrap4d
killing 1426823
Sleeping for 2 seconds
Killing ovs
sleeping for 2 seconds
Removing any vhost users from /tmp
Setting PATH
OS and Version details...
Fedora : 33


Updated Environment Variables ...
SDE_INSTALL: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install
LIBRARY_PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:
LD_LIBRARY_PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/lib/:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/install/lib64:/usr/local/lib:/usr/local/lib64
PATH: /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/sbin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/sbin:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/bin:/var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/deps/sbin

starting ovs
set hugepages
2048
2048
run infrap4d
check if infrap4d is running
root 1427758 1 0 09:44 ? 00:00:00 /var/jenkins_agent/workspace/nex-epgsw-ipdkci/GitHub_MultiBranch_Pipelines/p4cp-ptf-tests/ipdk-recipe/install/sbin/infrap4d
infrap4d Running
Using packet manipulation module: ptf.packet_scapy
7/13/2023,9:44:53  PASSED: gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-0,port-type:LINK"
7/13/2023,9:44:53  PASSED: gnmi-ctl set "device:virtual-device,name:net_vhost1,host-name:host2,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-1,port-type:LINK"
7/13/2023,9:44:55  PASSED: gnmi-ctl get verified for device:virtual-device,name:net_vhost0,host-name:host1,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-0,port-type:LINK
7/13/2023,9:44:55  PASSED: gnmi-ctl get verified for device:virtual-device,name:net_vhost1,host-name:host2,device-type:VIRTIO_NET,queues:1,socket-path:/tmp/vhost-user-1,port-type:LINK
7/13/2023,9:44:55  INFO: p4c-dpdk -I p4include -I p4include/dpdk --p4v=16 --p4runtime-files             common/p4c_artifacts/l3em_indirect_counter_pna/p4Info.txt -o common/p4c_artifacts/l3em_indirect_counter_pna/pipe/l3em_indirect_counter_pna.spec --arch pna --bf-rt-schema common/p4c_artifacts/l3em_indirect_counter_pna/bf-rt.json --context             common/p4c_artifacts/l3em_indirect_counter_pna/pipe/context.json common/p4c_artifacts/l3em_indirect_counter_pna/l3em_indirect_counter_pna.p4
7/13/2023,9:44:56  PASSED: p4c-dpdk -I p4include -I p4include/dpdk --p4v=16 --p4runtime-files             common/p4c_artifacts/l3em_indirect_counter_pna/p4Info.txt -o common/p4c_artifacts/l3em_indirect_counter_pna/pipe/l3em_indirect_counter_pna.spec --arch pna --bf-rt-schema common/p4c_artifacts/l3em_indirect_counter_pna/bf-rt.json --context             common/p4c_artifacts/l3em_indirect_counter_pna/pipe/context.json common/p4c_artifacts/l3em_indirect_counter_pna/l3em_indirect_counter_pna.p4
7/13/2023,9:44:56  PASSED: ovs_pipeline_builder --p4c_conf_file=l3em_indirect_counter_pna.conf             --bf_pipeline_config_binary_file=l3em_indirect_counter_pna.pb.bin
7/13/2023,9:44:57  PASSED: p4rt-ctl set pipe: common/p4c_artifacts/l3em_indirect_counter_pna/l3em_indirect_counter_pna.pb.bin
7/13/2023,9:44:57  INFO: Scenario : MainControlImpl_ipv4_host
7/13/2023,9:44:57  INFO: Adding MainControlImpl_ipv4_host rules
7/13/2023,9:44:57  PASSED: p4rt-ctl add entry: hdr.ipv4.dstAddr=1.1.1.1,action=MainControlImpl.send_with_bytes_incr(0,1)
7/13/2023,9:44:57  PASSED: p4rt-ctl add entry: hdr.ipv4.dstAddr=2.2.2.2,action=MainControlImpl.send_with_bytes_incr(1,2)
7/13/2023,9:44:57  INFO: Get flow dump table entry
7/13/2023,9:44:57  INFO: Verify each table entry
7/13/2023,9:44:57  PASSED: The entry "table=MainControlImpl.ipv4_host hdr.ipv4.dstAddr=0x01010101 actions=MainControlImpl.send_with_bytes_incr(port=0x00000000idx=0x0001)" in flow dump is verified
7/13/2023,9:44:57  PASSED: The entry "table=MainControlImpl.ipv4_host hdr.ipv4.dstAddr=0x02020202 actions=MainControlImpl.send_with_bytes_incr(port=0x00000001idx=0x0002)" in flow dump is verified
7/13/2023,9:44:57  INFO: delete table entries
7/13/2023,9:44:57  INFO: Deleting MainControlImpl_ipv4_host rules
7/13/2023,9:44:57  PASSED: p4rt-ctl del entry: hdr.ipv4.dstAddr=1.1.1.1
7/13/2023,9:44:57  PASSED: p4rt-ctl del entry: hdr.ipv4.dstAddr=2.2.2.2
7/13/2023,9:44:57  INFO: Test has PASSED



====================
Summary
====================
DPDK_Flow_Dump_with_LNT : PASSED
DPDK_Flow_Dump_with_PNA_indirect_counter : PASSED

Ran 2 test(s), 2 Passed, 0 Failed.

hirentha reviewed (approved):
LGTM.
